apps/spruce/src/pages/configurePatch/configurePatchCore/useConfigurePatch/index.ts

```diff
@@ -1,169 +1,58 @@
-  | { type: "setSelectedBuildVariants"; buildVariants: string[] }
-  | { type: "setPatchParams"; params: ParameterInput[] }
-  | { type: "setSelectedBuildVariantTasks"; variantTasks: VariantTasksState }
-  | { type: "setSelectedTab"; tabIndex: number }
-  | {
-      type: "updatePatchData";
-      description: string;
-      buildVariants: string[];
-      params: ParameterInput[];
-      variantTasks: VariantTasksState;
-      aliases: AliasState;
-    }
-  | {
-      type: "setSelectedAliases";
-      aliases: AliasState;
-    };
-
-const initialState = ({ selectedTab = 0 }: { selectedTab: number }) => ({
-  description: "",
-  selectedAliases: {},
-  selectedBuildVariants: [],
-  selectedBuildVariantTasks: {},
-  patchParams: null,
-  selectedTab,
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  disableBuildVariantSelect: tabToIndexMap[selectedTab] === PatchTab.Tasks,
-});
-
-const reducer = (state: ConfigurePatchState, action: Action) => {
-  switch (action.type) {
-    case "setDescription":
-      return {
-        ...state,
-        description: action.description,
-      };
-    case "setSelectedBuildVariants":
-      return {
-        ...state,
-        selectedBuildVariants: action.buildVariants.sort((a, b) =>
-          b.localeCompare(a),
-        ),
-      };
-    case "setSelectedBuildVariantTasks":
-      return {
-        ...state,
-        selectedBuildVariantTasks: action.variantTasks,
-      };
-    case "setSelectedAliases":
-      return {
-        ...state,
-        selectedAliases: action.aliases,
-      };
-    case "setPatchParams":
-      return {
-        ...state,
-        patchParams: omitTypename(action.params),
-      };
-    case "setSelectedTab": {
-      let tab = indexToTabMap.indexOf(PatchTab.Tasks);
-      if (action.tabIndex !== -1 && action.tabIndex < indexToTabMap.length) {
-        tab = action.tabIndex;
-      }
-      return {
-        ...state,
-        selectedTab: tab,
-        disableBuildVariantSelect:
-          indexToTabMap[action.tabIndex] !== PatchTab.Tasks,
-      };
-    }
-    case "updatePatchData":
-      return {
-        ...state,
-        description: action.description,
-        selectedBuildVariants: action.buildVariants,
-        patchParams: omitTypename(action.params),
-        selectedBuildVariantTasks: action.variantTasks,
-        selectedAliases: action.aliases,
-      };
-
-    default:
-      throw new Error("Unknown action type");
-  }
-};
-
-const indexToTabMap = [PatchTab.Tasks, PatchTab.Changes, PatchTab.Parameters];
-
-const tabToIndexMap = {
-  [PatchTab.Tasks]: 0,
-  [PatchTab.Changes]: 1,
-  [PatchTab.Parameters]: 2,
-};
+import { ConfigurePatchPageTabs } from "types/patch";
+import { indexToTabMap, tabToIndexMap } from "./constants";
+import { ConfigurePatchState, initialState, reducer } from "./state";
+import { AliasState, VariantTasksState } from "./types";
+import { initializeAliasState, initializeTaskState } from "./utils";
```

Comment:
It seems like Configure Patch page used to have some logic where if you visit an invalid tab, it would take you to the tasks tab. I think that might've been lost somewhere in this PR!

Reply:
Great catch!